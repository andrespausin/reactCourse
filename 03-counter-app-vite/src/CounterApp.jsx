import { useState } from "react";

// La idea de manejar eventos fuera del componente es que no se re-renderice el componente al hacer click en el botÃ³n
    const getEvent = (event) => {
        console.log(event);
    }

export const CounterApp = ({ value }) => {
    const [counter, setCount] = useState(value);

    const handleAdd = () => {
        setCount(counter+1)
    }

    const handleSubstract = () => {
        setCount(counter-1)
    }

    const handleReset = () => {
        setCount(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ `Contador: ${counter}` }</h2>

            <div className="btn-container">
                <button onClick={handleAdd}>+1</button>
                <button onClick={handleSubstract}>-1</button>
                <button onClick={handleReset}>Reset</button>
            </div>
            
        </>
    )
}