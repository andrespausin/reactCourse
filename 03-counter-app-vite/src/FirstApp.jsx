import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {
    return (
        <>
            <h3>{title}</h3>
            <p>{subtitle}</p>
            {/* <p>{JSON.stringify(newMessage)}</p>  */}
        </>
        
    )
}

// Prptypes para validar las propiedades del componente
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}