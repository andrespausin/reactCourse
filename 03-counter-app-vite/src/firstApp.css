html, body {
    margin: 0;
    padding: 0;
    font-family: Arial, sans-serif;
    background-color: black;
    color: white;
}

div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

button {
    padding: 10px 20px;
    font-size: 16px;
    color: white;
    background-color: #007BFF;
    border: none;
    border-radius: 5px;
    cursor: pointer;
}

button:hover {
    background-color: #0056b3;
}