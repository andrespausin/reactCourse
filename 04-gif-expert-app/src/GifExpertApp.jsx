import { useEffect, useState } from "react"

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['dragon ball', 'one punch man'])

    const onAddCategory = () => {
        setCategories([...categories, 'Hola mundo'])
    }

    return (
        <>
            {/* Titulo */}
            <h1>Gif Expert App</h1>

            {/* Input */}

            {/* .Listado de Gif */}
            <button onClick={onAddCategory}>Agregar</button>
            <ol>
                {
                    categories.map((category, index) => {
                        return <li key={index}>{category}</li>
                    })
                }
            </ol>
                {/* Gif Item */}
        </>
    )
}