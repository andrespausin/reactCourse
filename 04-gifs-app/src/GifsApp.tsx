import { useState } from "react"
import { GifContainer } from "./gifs/components/GifContainer"
import { mockGifs } from "./mock-data/data"
import { CustomHeader } from "./shared/components/CustomHeader"
import { PreviousSearches } from "./shared/components/PreviousSearches"
import { SearchBar } from "./shared/components/SearchBar"

export const GifsApp = () => {
    const [previousSearches, setPreviousSearches] = useState<string[]>(["dragon ball"])

    const handleTermClicked = (term: string) => {
        console.log("Term clicked:", term)
    }

    const handleSearch = (searchTerm: string) => {
        // 2. El string ni empieza ni termina con espacios
        const search = searchTerm.trim().toLowerCase();

        // 1. El string no puede estar vacio
        if (search.trim() === "") return;


        // 3. No hay búsquedas repetidas y solo se guardan las últimas 8 búsquedas
        setPreviousSearches((prevSearches) => {
            if (prevSearches.includes(search)) {
                return prevSearches;
            }
            if (prevSearches.length >= 8) {
                return [search, ...prevSearches.slice(0, 7)];
            }
            return [search, ...prevSearches];
        });
    }

    return (
        <>
            {/* Custom Header */}
            <CustomHeader title="Gifs App" description="Descubre y comparte el gif perfecto" />

            {/* Search Bar y busquedas previas */}
            <SearchBar placeholder="Busca lo que + desees" onQuery={handleSearch} />

            <PreviousSearches previousSearches={previousSearches} onSearchClicked={handleTermClicked} />

            {/* Gifs */}
            <GifContainer gifs={mockGifs} />
        </>
    )
}
