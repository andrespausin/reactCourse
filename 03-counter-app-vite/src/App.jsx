// import { FirstApp } from "./FirstApp"
import { CounterApp } from "./CounterApp"

const firstTitle = 'Soy una propiedad que viene desde el componente padre'
const firstSubtitle = 'Soy un subtitulo que viene desde el componente padre'

export const App = () => {
    
    return (
        <>
            <CounterApp value={10} />
        </>
    )
}