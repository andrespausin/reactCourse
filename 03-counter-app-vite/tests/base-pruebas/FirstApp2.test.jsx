import {render, screen} from '@testing-library/react';
import {FirstApp} from '../../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
    const name = 'Goku';
    const subtitle = 'Soy un subtitulo por defecto desde FirstApp';
    
    test("Debe de hacer match con el snapshot", () => {
        
        const {container} = render(<FirstApp name={name} />);
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el mensaje "Hola, soy Goku" en un H1', () => {

        render(<FirstApp name={name} />);
        expect(screen.getByText(name)).toBeTruthy();
    })

    test('Debe mostrar el titutlo en un h1', () => {

        render(<FirstApp name={name} title="Hola mundo" />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain('Hola mundo');
    }) 

    test('Debe de mostrar el subtitulo enviado por props', () => {

        render(<FirstApp name={name} subtitle={subtitle} />);
        expect(screen.getAllByText(subtitle)).toBeTruthy();
    })
})