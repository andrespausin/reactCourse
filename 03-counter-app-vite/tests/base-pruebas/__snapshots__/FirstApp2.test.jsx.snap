// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Soy un titulo por defecto desde FirstApp
  </h1>
  <p>
    Soy un subtitulo por defecto desde FirstApp
  </p>
  <p
    id="name"
  >
    Goku
  </p>
</div>
`;
